### PR TITLE
refactor poster image logic to work with urls

### PR DESCRIPTION
**Pull request by akash1810**
Previously, we were creating a read stream from a url, which is not possible. Now, we download the file to `/tmp/videoid.jpg` then create a read stream from that.

cc @fredex42 @clloyd @Reettaphant

### CODE REVIEW

File: CDS/scripts/js_utils/youtube/youtube-upload-lib.js [hunk 1]
```diff
@@ -98,19 +99,27 @@ function addPosterImageIfExists(connection, videoId, youtubeClient, account) {
...
-                    media: {
-                        body: fs.createReadStream(filename)
-                    }
-                };
-
-                youtubeClient.thumbnails.set(payload, (err, result) => err ? reject(err) : resolve(result));
-            }).catch(err => {
-                reject(err);
-            });
+            downloadPosterImage(posterImage.value, `${POSTER_IMAGE_DOWNLOAD_DIR}/${videoId}.jpg`)
```
Comment (clloyd) [inline CDS/scripts/js_utils/youtube/youtube-upload-lib.js]: Can we presume jpg? this is the grid

Comment (akash1810) [inline CDS/scripts/js_utils/youtube/youtube-upload-lib.js]: does `createReadStream` care about file extension? Presumably it reads it as bytes? I might just omit it entirely...

**Comment by clloyd** [comment]
Approved but should wait for @fredex42 to have a look

File: CDS/scripts/js_utils/youtube/youtube-upload-lib.js [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
+const Promise = require('promise');
+const fs = require('fs');
+const youtubeAuth = require('./youtube-auth.js');
+const googleapis = require('googleapis');
+const OAuth2 = googleapis.auth.OAuth2;
+const dataStore = require('../Datastore');
+const https = require('https');
 
 const YOUTUBE_API_VERSION = 'v3';
+const POSTER_IMAGE_DOWNLOAD_DIR = '/tmp';
```
Comment (fredex42) [inline CDS/scripts/js_utils/youtube/youtube-upload-lib.js]: would prefer that this is a default that can be changed in-route - e.g., when we deploy to AWS would want this on instance storage if it's available

**Comment by fredex42** [comment]
approved, have added a couple of lines of doc :)

**Pull request merged**